src/lib/lmdb_storage/block.ml

```diff
@@ -0,0 +1,208 @@
+      Helper.test_with_libp2p_helper ~logger ~handle_push_message
+        (fun conf_dir helper ->
+          let%bind me = generate_random_keypair helper in
+          let maddr =
+            multiaddr_to_libp2p_ipc
+            @@ Mina_net2.Multiaddr.of_string "/ip4/127.0.0.1/tcp/12878"
+          in
+          let libp2p_config =
+            Libp2p_ipc.create_libp2p_config
+              ~private_key:(Mina_net2.Keypair.secret me)
+              ~statedir:conf_dir ~listen_on:[ maddr ] ~external_multiaddr:maddr
+              ~network_id:"s" ~unsafe_no_trust_ip:true ~flood:false
+              ~direct_peers:[] ~seed_peers:[] ~known_private_ip_nets:[]
+              ~peer_exchange:true ~peer_protection_ratio:0.2 ~min_connections:20
+              ~max_connections:40 ~validation_queue_size:250
+              ~gating_config:empty_libp2p_ipc_gating_config ?metrics_port:None
+              ~topic_config:[] ()
+          in
+          let%bind _ =
+            Helper.do_rpc helper
+              (module Libp2p_ipc.Rpcs.Configure)
+              (Libp2p_ipc.Rpcs.Configure.create_request ~libp2p_config)
+            >>| Or_error.ok_exn
+          in
+          f conf_dir helper )
+
+    let send_and_receive ~helper ~reader ~db breadcrumb =
+      let body = Breadcrumb.block breadcrumb |> Mina_block.body in
+      let body_ref = Staged_ledger_diff.Body.compute_reference body in
+      [%log info] "Sending add resource" ;
+      Mina_net2.For_tests.Helper.send_add_resource
+        ~tag:Staged_ledger_diff.Body.Tag.Body ~body helper ;
+      [%log info] "Waiting for push message" ;
+      let%map id_ = Pipe.read reader in
+      let id = match id_ with `Ok a -> a | _ -> failwith "unexpected" in
+      [%log info] "Push message received" ;
+      [%test_eq: String.t] (Consensus.Body_reference.to_raw_string body_ref) id ;
+      [%test_eq:
+        ( Mina_block.Body.t
+        , [ `Invalid_structure of Error.t | `Non_full | `Tx_failed ] )
+        Result.t] (Ok body) (read_body db body_ref)
+
+    let%test_unit "Write many blocks" =
+      let n = 300 in
+      Quickcheck.test (gen_breadcrumb ~verifier ()) ~trials:1
+        ~f:(fun make_breadcrumb ->
+          let frontier = create_frontier () in
+          let root = Full_frontier.root frontier in
+          let reader, writer = Pipe.create () in
+          with_helper ~writer (fun conf_dir helper ->
+              let db =
+                create (String.concat ~sep:"/" [ conf_dir; "block-db" ])
+              in
+              let%bind breadcrumb = make_breadcrumb root in
+              let%bind () = send_and_receive ~db ~helper ~reader breadcrumb in
+              Quickcheck.test
+                (String.gen_with_length 1000
+                   (* increase to 1000000 to reach past mmap size of 256 MiB*)
+                   Base_quickcheck.quickcheck_generator_char ) ~trials:n
+                ~f:(fun _ ->
+                  let body = Breadcrumb.block breadcrumb |> Mina_block.body in
```

Comment:
I used already created breadcrumb instead of generated string, is there any way to convert string to ~body ?